src/Storages/ObjectStorageQueue/StorageObjectStorageQueue.cpp

```diff
@@ -962,6 +968,7 @@ void StorageObjectStorageQueue::alter(
         LOG_TEST(log, "New settings: {}", new_metadata.settings_changes->formatForLogging());
 
         /// Alter settings which are stored in keeper.
+        std::lock_guard lock_startup_mutex(startup_mutex);
```

Comment:
here as well, but here we can also throw an exception if `!startup_called` or just `chassert(startup_called)`